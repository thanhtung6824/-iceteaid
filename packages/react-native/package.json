{
  "name": "iceteaid-react-native",
  "version": "0.0.105-alpha.0",
  "description": "> TODO: description",
  "author": "Tung <thanhtung6824@gmail.com>",
  "homepage": "",
  "license": "MIT",
  "main": "dist/index.js",
  "dependencies": {
    "iceteaid-core": "^0.0.54-alpha.0",
    "rxjs": "7.0.0-beta.5",
    "whatwg-url": "8.1.0"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.11.5",
    "@babel/preset-typescript": "^7.10.4",
    "@testing-library/react-native": "^7.0.2",
    "@types/jest": "^26.0.14",
    "@types/whatwg-url": "^8.0.0",
    "babel-loader": "^8.1.0",
    "browser-env": "^3.3.0",
    "eslint-import-resolver-typescript": "^2.3.0",
    "eslint-plugin-import": "^2.22.1",
    "import-fresh": "^3.2.1",
    "jest": "^26.4.2",
    "metro-react-native-babel-preset": "^0.63.0",
    "react": "^16.13.1",
    "react-native": "^0.63.2",
    "react-native-webview": "^10.9.0",
    "react-test-renderer": "^16.13.1",
    "ts-jest": "^26.4.0",
    "webpack-node-externals": "^2.5.2"
  },
  "gitHead": "dec33862b3de41144e698acf4736f23726bfb959"
}
