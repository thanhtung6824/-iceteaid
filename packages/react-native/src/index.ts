import { SdkBase, sdkBuilder } from 'iceteaid-core';
import { NativeTransporter } from './native-transporter';
import { NativeIframe } from './native-iframe';
import { FC } from 'react';

export class SdkNative extends SdkBase {
    public get IFrame(): FC {
        return (this.iframe as unknown as NativeIframe).IFrame;
    }
}

export const IceteaId = sdkBuilder(SdkNative, {
    target: 'react-native',
    baseUrl: 'http://3k.theydont.work',
    Transporter: NativeTransporter,
    Iframe: NativeIframe,
});
