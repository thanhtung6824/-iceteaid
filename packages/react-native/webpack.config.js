const path = require('path');
const nodeExternals = require('webpack-node-externals');

module.exports = {
    mode: 'production',
    entry: path.join(__dirname, 'src/index.ts'),
    target: 'node',
    externals: [
        nodeExternals(),
    ],
    module: {
        rules: [
            {
                test: /\.(ts|tsx)?$/,
                exclude: /node_modules/,
                include: path.resolve(__dirname, 'src'),
                use: [
                    {
                        loader: 'ts-loader',
                    },
                ]
            },
        ],
    },
    resolve: {
        extensions: [ '.tsx', '.ts', '.js' ],
    },
    output: {
        filename: 'index.js',
        path: path.resolve(__dirname, 'dist'),
        libraryTarget: 'umd',
        library: '',
    },
};
