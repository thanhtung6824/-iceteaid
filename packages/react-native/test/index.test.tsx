/**
 * @jest-environment jsdom
 */
import React from 'react';
import { createSdk, createNativeIframe } from './mocks';
import { render } from '@testing-library/react-native';

it('Match snapshot', () => {
    const iceteaId = createNativeIframe();
    const component = render(<iceteaId.IFrame />);
    expect(component.toJSON()).toMatchSnapshot();
});

it ('Get Iframe will be same with Sdk', () => {
    const { IceteaId } = createSdk();
    const iceteaId = new IceteaId('xxx');
    expect(iceteaId.IFrame).toStrictEqual((iceteaId.iframe as any).IFrame);
});

it ('Global variable were defined', () => {
    delete (global as any)['URL'];
    delete (global as any).URLSearchParams;
    delete (global as any).atob;
    delete (global as any).btoa;
    expect((global as any).URL).toEqual(undefined);
    expect((global as any).URLSearchParams).toEqual(undefined);
    expect((global as any).atob).toEqual(undefined);
    expect((global as any).btoa).toEqual(undefined);
    createSdk();
    expect((global as any).URL).toBeDefined();
    expect((global as any).URLSearchParams).toBeDefined();
    expect((global as any).atob).toBeDefined();
    expect((global as any).btoa).toBeDefined();
});
