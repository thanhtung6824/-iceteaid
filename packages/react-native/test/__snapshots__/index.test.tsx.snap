// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
      },
      Object {
        "elevation": -10000,
        "zIndex": -10000,
      },
    ]
  }
  testID="qwerty"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={true}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onNavigationStateChange={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "http://3k.theydont.work",
        }
      }
      startInLoadingState={true}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      testID="abcxyz"
      useSharedProcessPool={true}
      userAgent="Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/603.1.23 (KHTML, like Gecko) Version/10.0 Mobile/14E5239e Safari/602"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
